server/apps/immich/src/api-v1/oauth/oauth.service.spec.ts

```diff
@@ -155,12 +164,15 @@ describe('OAuthService', () => {
       jest.spyOn(sut['logger'], 'debug').mockImplementation(() => null);
       jest.spyOn(sut['logger'], 'log').mockImplementation(() => null);
       userRepositoryMock.getByEmail.mockResolvedValue(null);
+      userRepositoryMock.getAdmin.mockResolvedValue(user);
```

Comment:
Why is there a call to `getAdmin` for a user that already exists? This test is about updating a user (not setting isAdmin: true), so I'm not sure why this call is happening here.

Reply:
Actually this test is about creating a user.
The oauth service checks if the user exists, and it gets a null `userRepositoryMock.getByEmail.mockResolvedValue(null);`
Then in attempts to create a user by doing `this.userCore.createUser`, which as part of the business logic checks if an admin already exists `userRepositoryMock.getAdmin.mockResolvedValue(user);`
I guess I was lazy here and didn't create an additional `userAdmin` resource to return here so it's different  than the one returned by the `create` mock (maybe it would be more readable, but still that getAdmin call would still be needed) 😬

Reply:
Oh, you're right. I must have misread the diff. I thought this for the update case.